python/sqlflow_playground/playground_server_design.md

```diff
@@ -0,0 +1,135 @@
+# SQLFlow Playground Server
+
+SQLFlow Playground Server exposes a REST API service that enables users to share
+the resources in one playground cluster. Users can take advantage of SQLFlow by
+installing a small [plugin](https://github.com/sql-machine-learning/pysqlflow)
+on her/his Jupyter Notebook.
+
+This service is used to extend the SQLFlow Playground's capability, especially
+when we need to manage the resource in the k8s cluster. We suppose the playground
+as a pure backend service (without Jupyter/JupyterHub) which provides machine 
+learning capability for some frontend. It clearly is not for those who just want
+to connect to the SQLFlow server in the playground through our built-in Jupyter 
+Notebook. Currently, this service is used to run our tutorials on [Aliyun
+DSW for Developer](https://dsw-dev.data.aliyun.com/) which behaves as a frontend
+of our playground.
+
+## The Architecture
+
+**SQLFlow Playground Server** is a side-car service of our playground cluster.
+Now, it is designed as an HTTP server which receives user login, creates DB
+resource, and so on. This server uses `kubectl` to manipulate the resource in
+the playground(a k8s cluster). It's in someway the gateway of the playground.
+As described in the below diagram, the interaction of the three subjects could
+be: Clients ask the playground server for some resource. The server authorizes
+the client and create the resource on the playground. The client connects to
+the SQLFlow server in the playground and does train/predict tasks using the
+created resource.
+
+```
+   ----------------run task--------------------------->
+   |                                                  |
+Clients <--> Playground Server <--> Playground[SQLFlow Server, MySQL Server...]
+```
+
+## Supported API
+
+Request URL path is composed by the prefix `/api/` and the api name, like:
+
+```url
+    https://playground.sqlflow.tech/api/heart_beat
+```
+This service always uses `HTTPS` and only accepts authorized clients
+by checking their certification file. So there is no dedicated api
+for user authentication.
+
+Currently supported API are:
+| name | method | params | description |
+| - | - | - | - |
+| create_db | POST | {"user_id": "id"} | create a DB for given user, json param |
+| heart_beat| GET  | user_id=id | report a heart beat of given client |
+
+
+## How to Use
+
+### For Service Maintainer
+The maintainer should [provide the playground cluster](../dev.md), and
+bootup a `SQLFlow Playground Server`.  The server should have privillege
+to access the `kubectl` command of the cluster.  To install the server,
+maintainer can use below command:
+```bash
+    mkdir $HOME/workspace
```

Comment:
please add `$` at the front of every bash command.

Reply:
why we should do that?